### DISCUSSION TITLE
Downloading a file through useCustom

### DISCUSSION THREAD

**Discussion by jmrfrosa** [refinedev/refine#2921] [Q&A] [answered]
Hey all,

I've recently run into an issue: I'm trying to implement a "Download" button per resource listing similar to an export feature. The data comes from my own backend in `xlsx` format and accepts the same filtering/sorting/pagination parameters as the listing page.

Hence, I've been trying to use `useCustom`, since it supports the above parameters while also letting me configure the headers. However, it seems my data always comes back corrupted, likely due to `useCustom` not allowing me to configure the `responseType` of the `httpClient` (which should be `blob`).

Implementing my own fetch query and specifying the correct `responseType` works totally fine and allows me to download the file – but forces me to rewrite the filter/sorting/pagination generators the data provider already implements.

Is there a way to get around this or somehow extend the hook to add this option?

**Comment by omeraplak** [comment, marked as answer]
Hey @jmrfrosa ,
Sorry for the late reply. Maybe you can implement it with `metaData`.
https://refine.dev/docs/faq/#how-i-can-override-specific-function-of-data-providers

**Reply by jmrfrosa** [reply]
Thanks @omeraplak - this looks useful. I ended up solving it in a similar way by duplicating and modifying the base `dataProvider` and `useCustom` to suit my needs. I'll attempt to reduce this duplication a bit with the method above.

**Reply by aunghtain** [reply]
@jmrfrosa can you share the code? I'm trying to implement download and it's not working. TIA
